test/ng/controllerSpec.js

```diff
@@ -57,6 +57,27 @@ describe('$controller', function() {
       expect(ctrl instanceof BarCtrl).toBe(true);
     });
 
+    it('should return true when having an existing controller', function() {
+      var FooCtrl = function($scope) { $scope.foo = 'foo'; },
+          scope = {},
+          ctrl;
+
+      $controllerProvider.register({FooCtrl: FooCtrl});
+
+      ctrl = $controller('FooCtrl', {$scope: scope});
+      expect(ctrl.has('FooCtrl')).toBe(true);
```

Comment:
It's `$controllerProvider` that has the `.has()` method. Not specific controller instances.
This test could look like:

```js
$controllerProvider.register('FooCtrl', noop);

expect($controllerProvider.has('FooCtrl')).toBe(true);
expect($controllerProvider.has('BarCtrl')).toBe(false);
```

Or if we wanted to be more thorough, something like:

```js
$controllerProvider.register('FooCtrl', noop);
$controllerProvider.register('BarCtrl', ['dep1', 'dep2', noop]);
$controllerProvider.register({
  'BazCtrl': noop,
  'QuxCtrl: ['dep1', 'dep2', noop]
});

expect($controllerProvider.has('FooCtrl')).toBe(true);
expect($controllerProvider.has('BarCtrl')).toBe(true);
expect($controllerProvider.has('BazCtrl')).toBe(true);
expect($controllerProvider.has('QuxCtrl')).toBe(true);

expect($controllerProvider.has('UnknownCtrl')).toBe(false);
```

And basically, the next test would be redundant.